### PR TITLE
feat(i18n,ui5): implement ui5 locale support

### PR DESCRIPTION
**Pull request by droshev**
feat(ui5): locale ui5

  BREAKING CHANGE: `FD_LOCALE_SIGNAL` now derives from `FD_LANGUAGE_SIGNAL` via linkedSignal (default `en` instead of `en-US`), and `FD_LANGUAGE_SIGNAL` auto-detects from `LOCALE_ID` at bootstrap instead of
  defaulting to English — disable with `{ provide: FD_LANGUAGE_AUTO_DETECT, useValue: false }`.

New APIs:
  - FD_LOCALE_SIGNAL — WritableSignal<string> derived from FD_LANGUAGE_SIGNAL via linkedSignal(). Auto-resets on language change, writable for locale overrides.
  - FD_LANGUAGE_AUTO_DETECT — InjectionToken<boolean> (default true). Controls whether FD_LANGUAGE_SIGNAL reads LOCALE_ID at bootstrap to pick the closest built-in language.
  - detectLanguage(localeId) — Maps a locale string (e.g., 'de', 'bg-BG') to the closest FdLanguage constant. Exact match → base language fallback → English.

Interface changes:
  - FdLanguage — Added optional locale?: string and name?: string metadata fields.
  - All 37 language constants — Augmented with locale and name (e.g., FD_LANGUAGE_GERMAN now has locale: 'de', name: 'Deutsch').

Bug fixes:
  - flattenTranslations() — Skips locale/name metadata keys so they don't leak into the translation key space.
  - chinese.ts — Re-exports from chinese_simplified.ts instead of duplicating the object.
  - tokens.ts — Removed misleading { optional: true } on LOCALE_ID inject.
  - detect-language.ts — Replaced mutable lazy let with const LOCALE_MAP.
  - patch-language.ts — Preserves locale/name metadata when patching languages.

Test coverage:
  - detect-language.spec.ts — 28 test cases covering exact match, base language fallback, region variants, script subtags, case insensitivity, edge cases. All 37 languages covered.
  - flatten-translations.spec.ts — 6 test cases including metadata skipping regression test.
  - tokens.spec.ts — Tests for FD_LANGUAGE_SIGNAL, FD_LOCALE_SIGNAL, FD_LANGUAGE_AUTO_DETECT, linkedSignal reset behavior.
  - patch-language.spec.ts — Tests for metadata preservation during patching.
  - languages.spec.ts — Tests all 37 language constants have locale and name.

Library Changes (libs/ui5-webcomponents-base/)

New APIs:
  - Ui5LanguageService — Watches FD_LOCALE_SIGNAL via effect(), calls UI5's setLanguage().
  - provideUi5LanguageBridge() — Returns EnvironmentProviders that activates Ui5LanguageService. App-level only.
  - New i18n secondary entry point with barrel exports.

Build fix:
  - project.json — Added "^build" to dependsOn so i18n:build completes before ui5-webcomponents-base:build on CI.
  - tsconfig.lib.json — Added paths mapping for @fundamental-ngx/i18n to dist/libs/i18n.

Tests:
  - ui5-language.service.spec.ts — Tests for bridge activation, setLanguage() reactivity.

Docs App Wiring

  - main.ts — Added @ui5/webcomponents-localization/dist/Assets.js (CLDR) and @ui5/webcomponents/dist/Assets.js (component i18n) imports.
  - app.config.ts — Added provideUi5LanguageBridge() so toolbar language changes propagate to UI5 components.
  - translations.ts — Fixed toolbar translations to include locale/name metadata (was causing FD_LOCALE_SIGNAL to fall back to 'en-US').

Documentation & Examples

New UI5 i18n docs page (libs/docs/ui5-webcomponents/i18n/):
  - Overview, Setup (2-step), How it works (signal flow diagram), Live example
  - Registered in docs-routes.ts and docs-data.json

New examples:
  - Auto-detect example — Shows LOCALE_ID, detected language, derived locale
  - Locale override example — Demonstrates linkedSignal reset behavior (German text + Japanese formatting)

  Updated docs:
  - Getting-started guide — Accurate LOCALE_ID description (not "browser locale").
  - Changing-translations guide — Decision table, auto-detect/locale-override/UI5 bridge sections, escaped @ characters
  - i18n-patterns.md (agents doc) — Decision guide, auto-detection, UI5 integration sections
  - libs/i18n/README.md — Architecture, UI5 integration, opt-out sections

### CODE REVIEW

File: libs/docs/i18n/getting-started/examples/getting-started-example.component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.example-bordered-card {
+    border: 1px solid var(--sapNeutralBorderColor);
+    border-radius: 4px;
+}
+
+.example-info-panel {
+    background: var(--sapBackgroundColor);
+    border-radius: 4px;
+}
```
Comment (InnaAtanasova) [inline libs/docs/i18n/getting-started/examples/getting-started-example.component.scss]: values should be in rem

File: libs/docs/i18n/changing-translations/examples/i18n-language-change-example/language-change-example.component.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.example-info-panel {
+    background-color: var(--sapBackgroundColor);
+}
+
+.example-bordered-card {
+    border: 1px solid var(--sapGroup_ContentBorderColor);
```
Comment (InnaAtanasova) [inline libs/docs/i18n/changing-translations/examples/i18n-language-change-example/language-change-example.component.scss]: should be in rem

File: libs/docs/i18n/changing-translations/examples/i18n-auto-detect-example/auto-detect-example.component.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.example-bordered-card {
+    border: 1px solid var(--sapGroup_ContentBorderColor);
```
Comment (InnaAtanasova) [inline libs/docs/i18n/changing-translations/examples/i18n-auto-detect-example/auto-detect-example.component.scss]: values in rem

File: libs/docs/i18n/getting-started/examples/getting-started-example.component.scss (hunk 1 above)
Comment (droshev) [reply libs/docs/i18n/getting-started/examples/getting-started-example.component.scss]: updated

File: libs/docs/i18n/changing-translations/examples/i18n-auto-detect-example/auto-detect-example.component.scss (hunk 3 above)
Comment (droshev) [reply libs/docs/i18n/changing-translations/examples/i18n-auto-detect-example/auto-detect-example.component.scss]: updated

**Comment by droshev** [comment]
> I see a few scss files added, is it possible to replace the classes with common-css classes? I saw in few examples you used them, would be better for consistency to use one approach.

yes, i tried but we need to bring the latest common-css as some of the classes are different now
